docs/amending-commits.md

```diff
@@ -0,0 +1,90 @@
+## How to amend commits
+
+Amending commits is useful to improve the quality of existing commits
+for example after code review or after an automatic tool, like GitCop,
+found problems in some commits.
+
+Note that this should not be done on commits that are in an official
+branch. It should be done on local only commits or on commits that are
+in a not yet merged pull request.
+
+### Improving commit messages
+
+* Note on the `commit-msg` hook
+
+After editing or creating a commit message, Git runs the commit-msg
+hook, if it has been set up. And this hook can modify the just edited
+or created commit message.
+
+This works even when editing the commit message did not change it. So
+the commit message could still end up being changed by the commit-msg
+hook.
+
+More information about Git Hooks is available on:
+
+http://git-scm.com/book/uz/v2/Customizing-Git-Git-Hooks
+
+* Improving the commit message of the last commit
+
+It's vey easy to improve the commit message of the last commit on the
+current branch. Just run:
+
+```
+$ git commit --amend
+```
+
+It will open an editor for you to change the commit message, and, when
+you save and close the editor, it will replace the last commit with a
+new one that has the improved commit message.
+
+This means that if you used the
+[setup_commit_msg_hook.sh script](dev/tools/hooks/setup_commit_msg_hook.sh)
+to automatically add some trailers if they are not there, then running
+`git commit --amend` and then saving and closing the editor is enough
+for the missing trailers if any to be added.
+
+* Improving the commit message of any commit
+
+It's also easy to improve the commit message of let's say the last 5
+commits on the current branch. Just run:
+
+```
+$ git rebase -i HEAD~5
+```
```

Comment:
may want to have a section on rebasing, and show how to rebase against new origin/master. (that's a very common case)